.github/workflows/phpunit-pgsql.yml

```
@@ -33,9 +21,34 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  changes:
+    runs-on: ubuntu-latest
+
+    outputs:
+      src: ${{ steps.changes.outputs.src}}
+
+    steps:
+      - uses: dorny/paths-filter@4512585405083f25c027a35db413c2b3b9006d50 # v2.11.1
+        id: changes
+        with:
+          filters: |
+            src:
+              - '.github/workflows/**'
+              - 'appinfo/**'
+              - 'lib/**'
+              - 'templates/**'
+              - 'tests/**'
+              - 'vendor/**'
+              - 'vendor-bin/**'
+              - '.php-cs-fixer.dist.php'
+              - 'composer.json'
+              - 'composer.lock'
+
   phpunit-pgsql:
     runs-on: ubuntu-latest
 
+    if: needs.changes.outputs.src == 'true'
```

```suggestion
    needs: changes
    if: needs.changes.outputs.src == 'true'
```

To wait for the changes job, otherwise result is not set.

---

I'm not 100% sure, but for the summary job, we may need "needs"  for both jobs to access the results.

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idoutputs

(the downstream jobs part)

---

> we may need "needs" for both jobs to access the results.

Yes, I just played around and came to the same result. it's fixed now.

---

Here is a run with affected changes:
https://github.com/nextcloud/spreed/actions/runs/6466794906
And here without:
https://github.com/nextcloud/spreed/actions/runs/6466828174